### DISCUSSION TITLE
Should adding some partial function in StaticConstructorGenerator? Or other implementations

### DISCUSSION THREAD

**Discussion by noliar** [HavenDV/DependencyPropertyGenerator#55] [Ideas]
Currently, In Avalonia, this library cannot generate  all the code related to DependencyProperty and RoutedEvent, etc. like `AddClassHandler`, `OverrideDefaultValue`. (Of course, `AddClassHandler`can change it to AddHanlder):
![image](https://github.com/user-attachments/assets/0aae29ee-0f68-4c72-ba3d-1422e622f585)

And a type can only  have one static constructor. 

If we manualy add both static constructor and OnXXChanged partial method, we will get  `error CS0111`

Should we add new partial method in StaticConstructorGenerator? like this:
```csharp
static partial void StartOfStaticConstructor();  // or other name
static partial void EndOfStaticConstructor(); // or other name
```
and adding manual code in partial method.

Or adding new attribute like `[PostStaticConstructorAttribute]` to a static method?

Or other implementations?

**Comment by HavenDV** [comment]
I think partial methods is the best solution.
In an ideal situation, I would of course do without using static constructors at all, now this is some kind of workaround, but my experience in Avalonia does not allow me to see other possible ways
